vsintegration/src/FSharp.Editor/LanguageService/SymbolHelpers.fs

```diff
@@ -86,10 +87,18 @@ module internal SymbolHelpers =
                             yield project
                             yield! project.GetDependentProjects() ]
                         |> List.distinctBy (fun x -> x.Id)
+
+                let onFound =
+                    fun _ _ symbolUseRange ->
+                        async { symbolUseRanges.Add symbolUseRange }
+
+                let! _ = getSymbolUsesInProjects (symbol, projectInfoManager, checker, projects, onFound, userOpName)
                     
-                let! _ = getSymbolUsesInProjects (symbol, projectInfoManager, checker, projects, (fun _ _ symbolUse -> async { symbolUses.Add symbolUse }), userOpName)
-                    
-                return toDict symbolUses }
+                // Distinct these down because each TFM will produce a new 'project'.
+                // Unless guarded by a #if define, symbols with the same range will be added N times
+                let symbolUseRanges = symbolUseRanges.ToArray() |> Array.distinct
```

Comment:
I think this is fine, but shouldn't we be trying to be distinct in FCS before doing it here?

Reply:
Since the FCS call is made per-project, I think we'd need another way to understand when we're yielding back what's ultimately a duplicate. For that to be in the FCS layer we'd need to introduce the concept of a project (or TFM?) into FCS. That feels kinda complicated to me. Since the caller of FCS ultimately has that additional context it feels more appropriate to have it here for now. Happy to approach it differently though, since it's annoying that all callers have to know about this.